### PR TITLE
fix issues #1193 and #1558 for Linux/Windows (LWJGL v3 hangs)

### PR DESCRIPTION
**Pull request by bob0bob**
TestAWTPanels hangs with LWJGL v3 #1558 /  jme3test.app.TestApplication hangs with LWJGL3 #1193

Made LWJGL3-JME to launch in the same way the LWJGL2-JME launches.  This always Windows and Linux to run, still needs testing on Max OS X

### CODE REVIEW

**Comment by bob0bob** [comment]
This is also related to issue  jme-lwjgl3 - bug for starting app #1514

**Comment by Jeddic** [comment]
From what I can tell you just reverted the changes? I can already tell you this breaks osx which is why it was changed in the first place.

**Comment by bob0bob** [comment]
I made more changes, based on the history of the changes and other issues about this issue.  It does end up checking to see if it is a Mac and do something else.   This is solution I'm using on my engine,  it needs to be tested but I worry there is not anyone to test for it.  I will cross my fingers for one to step up to test.

**Comment by pspeed42** [comment]
We should try to come up with code that doesn't use "magic guess the OS" string checks.  Even if that means changing the semantics of what start() means.

**Comment by stephengold** [comment]
Sorry if this is a stupid question, but why use `mainThread.run()` instead of `mainThread.start()`?

**Comment by bob0bob** [comment]
I found a way to check from inside LWJGL3,  they have a call to return a ENUM for the 3 types of hardware that is support.  I will put that in but last night my eclipse build started to mess up and I couldn't get it to run in debug to verify it.

The enum returns,  WINDOWS, LINUX or MAC.

They even have a class to verify it is running on the main thread and return false is not, so you can throw an exception.

**Comment by Jeddic** [comment]
I just confirmed that this crashes on OSX.

**Comment by bob0bob** [comment]
I made changes to remove the string comparison.  This does fix the Windows and Linux.   Jeddic, if you have a MAC,  Please look into a solution for Mac.  Looking into the JME-LWJGL2 library the create() method ends up doing the same for the Mac.

This is moving forward to getting LWJGL3 functioning.   I'm not sure what "CRASH" means,  because it does the run() call on the current thread.   If anything this is not a solution for Mac, but for other 2 OS.

**Comment by stephengold** [comment]
@Jeddic next time, please copy/paste the stack trace

**Comment by Jeddic** [comment]
```
SEVERE: Uncaught exception thrown in Thread[jME3 Main,5,main]
java.lang.ExceptionInInitializerError
	at org.lwjgl.glfw.GLFW.glfwInit(GLFW.java:1046)
	at com.jme3.system.lwjgl.LwjglWindow.createContext(LwjglWindow.java:197)
	at com.jme3.system.lwjgl.LwjglWindow.initInThread(LwjglWindow.java:515)
	at com.jme3.system.lwjgl.LwjglWindow.run(LwjglWindow.java:650)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.IllegalStateException: GLFW may only be used on the main thread. This check may be disabled with Configuration.GLFW_CHECK_THREAD0.
	at org.lwjgl.glfw.EventLoop.<clinit>(EventLoop.java:30)
	... 5 more
```
There you go.

**Comment by Jeddic** [comment]
So I tested the current version and it is close, but calling mainthread.run() won't work. There isn't a runnable assigned so it immediately returns and closes the application. Just calling run() directly works for me.

**Comment by bob0bob** [comment]
Jeddic,  thanks.    I but the run() call back for MacOS.   This will not solve the issue for Mac and AWT but will for Linux and Windows.

**Comment by Jeddic** [comment]
Yah, fixing it for osx will probably take changing the canvas to use the lwjgl-awt project.  Let me know if there is anything else that needs to be tested.

**Comment by stephengold** [comment]
If this PR doesn't solve the issue for macOS, then why not remove the `if (Platform.get() == Platform.MACOSX) {` clause?

Also, our preferred Java style specifies that braces are used with `if` statements, even when the body is empty or contains only a single statement, so the `if (waitfor)` clause(s) need braces.

**Comment by Jeddic** [comment]
Well, this pull request should fix the issue for windows / linux. The clause is there specifically so it doesn't break macOS in the process.

**Comment by bob0bob** [comment]
I added the braces for the correct coding.

We have to leave in the check for MAC,  because the Mac needs the run() call.  That call is blocking and non-returning.   But with the thread().start(),  that is a non-blocking call that will allow the return so the rest of the code gets executed after the call to app.start() in main.

**Comment by stephengold** [comment]
Rather than slog through another round of review, I fixed a couple minor issues and integrated the PR.

Thank you for your contribution to the Engine!

**Comment by stephengold** [comment]
Now TestAWTPanels is broken in a new and different way. See issue #1558 for details.

File: jme3-lwjgl3/src/main/java/com/jme3/system/lwjgl/LwjglWindow.java [hunk 1]
```diff
@@ -477,10 +479,21 @@ public void create(boolean waitFor) {
...
+        if (Platform.get() == Platform.MACOSX) {
+            // NOTE: this is required for Mac OS X!
+            mainThread = Thread.currentThread();
+            mainThread.setName("jME3 Main");
+            if (waitFor) {
+                LOGGER.warning("create(true) is not supported for macOS!");
+            }
+            run();
+        } else {
+            new Thread(this, "jME3 Main").start();
```
Comment (Ali-RS) [inline jme3-lwjgl3/src/main/java/com/jme3/system/lwjgl/LwjglWindow.java:491]: A bit late to the party!

@bob0bob `mainThread` should be initialized to newly create thread inside `else` statement.

**Comment by bob0bob** [comment]
Okay, I checked in the changes and tested it against the example from the forum where it is hanging.

**Comment by bob0bob** [comment]
Do I start a new merge request to get the changes over since it has already been merged.

**Comment by Ali-RS** [comment]
No need for that, it is already solved in #1722.

**Pull request merged**